presto-main/src/test/java/com/facebook/presto/cost/TestSimpleFilterProjectSemiJoinStatsRule.java

```diff
@@ -0,0 +1,239 @@
+    {
+        tester.close();
+        tester = null;
+    }
+
+    @Test
+    public void testFilterPositiveSemiJoin()
+    {
+        tester.assertStatsFor(pb -> {
+            Symbol a = pb.symbol("a", BIGINT);
+            Symbol b = pb.symbol("b", BIGINT);
+            Symbol c = pb.symbol("c", BIGINT);
+            Symbol semiJoinOutput = pb.symbol("sjo", BOOLEAN);
+            return pb.filter(
+                    semiJoinOutput.toSymbolReference(),
+                    pb.semiJoin(
+                            pb.values(a, b),
+                            pb.values(c),
+                            a,
+                            c,
+                            semiJoinOutput,
+                            Optional.empty(),
+                            Optional.empty(),
+                            Optional.empty()));
+        })
+                .withSourceStats(filter -> filter.getSources().get(0).getSources().get(0), PlanNodeStatsEstimate.builder()
+                        .setOutputRowCount(1000)
+                        .addSymbolStatistics(new Symbol("a"), aStats)
+                        .addSymbolStatistics(new Symbol("b"), bStats)
+                        .build())
+                .withSourceStats(filter -> filter.getSources().get(0).getSources().get(1), PlanNodeStatsEstimate.builder()
+                        .setOutputRowCount(2000)
+                        .addSymbolStatistics(new Symbol("c"), cStats)
+                        .build())
+                .check(check -> {
+                    check.outputRowsCount(180)
+                            .symbolStats("a", assertion -> assertion.isEqualTo(expectedFilteredPositiveAStats))
+                            .symbolStats("b", assertion -> assertion.isEqualTo(bStats))
+                            .symbolStatsUnknown("c")
+                            .symbolStatsUnknown("sjo");
+                });
+    }
+
+    @Test
+    public void testFilterPositiveNarrowingProjectSemiJoin()
+    {
+        tester.assertStatsFor(pb -> {
+            Symbol a = pb.symbol("a", BIGINT);
+            Symbol b = pb.symbol("b", BIGINT);
+            Symbol c = pb.symbol("c", BIGINT);
+            Symbol semiJoinOutput = pb.symbol("sjo", BOOLEAN);
+            return pb.filter(
+                    expression("sjo"),
+                    pb.project(Assignments.identity(semiJoinOutput, a),
+                            pb.semiJoin(
+                                    pb.values(a, b),
+                                    pb.values(c),
+                                    a,
+                                    c,
+                                    semiJoinOutput,
+                                    Optional.empty(),
+                                    Optional.empty(),
+                                    Optional.empty())));
+        })
+                .withSourceStats(filter -> filter.getSources().get(0).getSources().get(0).getSources().get(0), PlanNodeStatsEstimate.builder()
```

Comment:
> `filter.getSources().get(0).getSources().get(0).getSources().get(0)`

Looks beautiful. 
What about assigning `pb.values(a, b)` some predefined identifier and using that instead?